pycm/pycm_multilabel_cm.py

```diff
@@ -0,0 +1,215 @@
+            except ValueError:
+                raise pycmMultiLabelError(INVALID_CLASS_NAME_ERROR)
+            actual_vector_temp = []
+            predict_vector_temp = []
+            for item1, item2 in zip(
+                    self.actual_vector_multihot, self.predict_vector_multihot):
+                actual_vector_temp.append(item1[class_index])
+                predict_vector_temp.append(item2[class_index])
+            cm = ConfusionMatrix(
+                actual_vector_temp,
+                predict_vector_temp,
+                sample_weight=self.weights)
+            self.classwise_cms[class_name] = cm
+            return cm
+        return self.classwise_cms[class_name]
+
+    def get_cm_by_sample(self, index):
+        """
+        Return confusion matrices based on samples.
+
+        :param index: sample index for confusion matrix
+        :type index: int
+        :return: confusion matrix corresponding to the given sample number
+        """
+        if index < 0 or index >= len(self.actual_vector):
+            raise pycmMultiLabelError(VECTOR_INDEX_ERROR)
+        if index not in self.samplewise_cms:
+            cm = ConfusionMatrix(
+                self.actual_vector_multihot[index],
+                self.predict_vector_multihot[index],
+                sample_weight=self.weights)
+            self.samplewise_cms[index] = cm
+            return cm
+        return self.samplewise_cms[index]
+
+    def __str__(self):
+        """
+        Multilabel confusion matrix object string representation method.
+
+        :return: representation as str
+        """
+        return self.__repr__()
+
+    def __repr__(self):
+        """
+        Multilabel confusion matrix object representation method.
+
+        :return: representation as str
+        """
+        return "pycm.MultiLabelCM(classes: " + str(self.classes) + ")"
+
+    def __len__(self):
+        """
+        Multilabel confusion matrix object length method.
+
+        :return: length as int
+        """
+        return len(self.classes)
+
+
+def __mlcm_vector_handler__(
+        mlcm,
+        actual_vector,
+        predict_vector,
+        sample_weight,
+        classes):
+    """
+    Handle multilabel object conditions for vectors.
+
+    :param mlcm: multilabel confusion matrix
+    :type mlcm: pycm.MultiLabelCM object
+    :param actual_vector: actual vector
+    :type actual_vector: python list or numpy array of any stringable objects
+    :param predict_vector: vector of predictions
+    :type predict_vector: python list or numpy array of any stringable objects
+    :param sample_weight: sample weights list
+    :type sample_weight: list
+    :param classes: ordered labels of classes
+    :type classes: list
+    :return: None
+    """
+    if not isinstance(actual_vector, (list, numpy.ndarray)) or not \
+            isinstance(predict_vector, (list, numpy.ndarray)):
+        raise pycmVectorError(VECTOR_TYPE_ERROR)
```

Comment:
I have some concerns about different types of errors in `MultiLabelCM`, sometimes `pycmVectorError` sometimes `pycmMultiLabelError`!!!
We should talk about this confusion

Reply:
Discussed privately.